sqlmesh/utils/metaprogramming.py

```diff
@@ -61,6 +61,14 @@ def _code_globals(code: types.CodeType) -> t.Dict[str, None]:
     return variables
 
 
+def _globals_match(obj1: t.Any, obj2: t.Any) -> bool:
+    return obj1 == obj2 or (
```

Comment:
do you want == or is? we don't want False and 0 to match right

Reply:
yeah, I guess we have this bug today because we used to compare the objects using `!=`:

```python
>>> False != 0
False
```

I'll fix this by checking the type first.

Reply:
@tobymao I fixed this [here](https://github.com/TobikoData/sqlmesh/compare/d6a6d639265c3e0695ad6d45b7127107c436a959..96e18c9592de55b9f154433f27d219b9b3e4b7dd). There was actually a bug where we wouldn't detect clashing definitions for these globals, because we returned from `build_env`'s `walk` too early due to the `name in visited` condition. I also fixed that bug and added a test.

Reply:
looks good